### ISSUE TITLE
Dependencies will be downloaded even if locally available

### ISSUE THREAD

**Issue by awvwgk** [qcscine/cmake#2]
It seems like I cannot convince CMake to use versions of the third-party dependencies like `gtest` when they are locally installed. To illustrate the problem I modified the `ImportGTest.cmake` with the following patch:

```patch
diff --git a/ImportGTest.cmake b/ImportGTest.cmake
index a43a690..72cb266 100644
--- a/ImportGTest.cmake
+++ b/ImportGTest.cmake
@@ -7,11 +7,12 @@ macro(import_gtest)
   # If the target already exists, do nothing
   if(NOT TARGET gtest)
     # Try to find the package locally
-    find_package(GTest 1.8.1 QUIET)
+    find_package(GTest 1.8.1)
     if(TARGET gtest)
       message(STATUS "GTest 1.8.1 found locally at ${GTest_DIR}.")
     else()
       # Download it instead
+      message(STATUS "GTest was not found in your PATH, starting download.")
       include(DownloadProject)
       download_project(
         PROJ                googletest
```

Invoking CMake for *e.g.* readuct with

```
cmake -GNinja -B_build -S. -DCMAKE_BUILD_TYPE=Release -DBUILD_SPARROW=ON
```

will lead to the following output:

```
-- Found GTest: /usr/lib/libgtest.so (Required is at least version "1.8.1") 
-- GTest was not found in your PATH, starting download.
```

This test was performed on a Manjaro Linux using [CMake 3.17.3](https://www.archlinux.org/packages/extra/x86_64/cmake/) and [gtest 1.8.1](https://www.archlinux.org/packages/community/x86_64/gtest/).

**Comment by nabbelbabbel** [comment]
Thanks for the feedback.

This `Import` file has already seen an update in our internal repository, using more of the existing `Find...` functionality CMake offers. This update should correctly detect the targets and not cause the error reported here.

However, it has to be noted, that even using CMake to search for GTest can fail for some distros. This is usually due to old CMake versions and missing CMake files in the official GTest packages given by the repositories. In these cases a download will still be performed.

(The latter does not or should not be the case for the specific versions you reported here.)

This issues will stay open until the next release has been made.

Cheers,
Jan

**Comment by awvwgk** [comment]
Thanks for the prompt reply. Could you provide a patch or maybe a short reference to the respective CMake functionality to allow me to fix this locally myself in the meantime?

**Comment by nabbelbabbel** [comment]
What the changes should boil down to, is using the same `find_package()` call, but then testing for the `GTest` targets described here: https://cmake.org/cmake/help/v3.17/module/FindGTest.html 
and also the corresponding GMock targets.
Finally then each SCINE Module has to link against those targets (GTest::GTest and GMock::Main) instead of the ones used now.

Sadly these types of changes require patches of most SCINE repos which is why I would be hesitant to generate those patch files at the moment.

Maybe you can just hack/delete the check for the actual target and make cmake assume that it was found and then use it.

**Comment by weymutht** [comment]
This has been fixed with release 4.0.0 of development-utils. Therefore, I close this issue.
